test/Kestrel.FunctionalTests/Http2/TlsTests.cs

```diff
@@ -0,0 +1,126 @@
+
+                return context.Response.WriteAsync("hello world " + context.Request.Protocol);
+            }, new TestServiceContext(LoggerFactory),
+            kestrelOptions =>
+            {
+                kestrelOptions.Listen(IPAddress.Loopback, 0, listenOptions =>
+                {
+                    listenOptions.Protocols = HttpProtocols.Http2;
+                    listenOptions.UseHttps(_x509Certificate2, httpsOptions =>
+                    {
+                        httpsOptions.SslProtocols = SslProtocols.Tls11 | SslProtocols.Tls12;
+                    });
+                });
+            }))
+            {
+                var connection = server.CreateConnection();
+                var sslStream = new SslStream(connection.Stream);
+                await sslStream.AuthenticateAsClientAsync(new SslClientAuthenticationOptions()
+                {
+                    TargetHost = "localhost",
+                    RemoteCertificateValidationCallback = (_, __, ___, ____) => true,
+                    ApplicationProtocols = new List<SslApplicationProtocol>() { SslApplicationProtocol.Http2, SslApplicationProtocol.Http11 },
+                    EnabledSslProtocols = SslProtocols.Tls11, // Intentionally less than the required 1.2
+                }, CancellationToken.None);
+
+                var reader = PipeReaderFactory.CreateFromStream(PipeOptions.Default, sslStream, CancellationToken.None);
+                await WaitForConnectionErrorAsync(reader, ignoreNonGoAwayFrames: false, expectedLastStreamId: 0, expectedErrorCode: Http2ErrorCode.INADEQUATE_SECURITY);
+            }
+        }
+
+        private async Task WaitForConnectionErrorAsync(PipeReader reader, bool ignoreNonGoAwayFrames, int expectedLastStreamId, Http2ErrorCode expectedErrorCode)
+        {
+            var frame = await ReceiveFrameAsync(reader);
+
+            if (ignoreNonGoAwayFrames)
+            {
+                while (frame.Type != Http2FrameType.GOAWAY)
+                {
+                    frame = await ReceiveFrameAsync(reader);
+                }
+            }
+
+            Assert.Equal(Http2FrameType.GOAWAY, frame.Type);
+            Assert.Equal(8, frame.Length);
+            Assert.Equal(0, frame.Flags);
+            Assert.Equal(0, frame.StreamId);
+            Assert.Equal(expectedLastStreamId, frame.GoAwayLastStreamId);
+            Assert.Equal(expectedErrorCode, frame.GoAwayErrorCode);
+        }
+
+        private async Task<Http2Frame> ReceiveFrameAsync(PipeReader reader)
+        {
+            var frame = new Http2Frame();
+
+            while (true)
+            {
+                var result = await reader.ReadAsync();
+                var buffer = result.Buffer;
+                var consumed = buffer.Start;
+                var examined = buffer.End;
+
+                if (buffer.IsEmpty && result.IsCompleted)
+                {
+                    throw new IOException("The reader completed without returning a frame.");
```

Comment:
Do you know why this is failing on [Travis](https://travis-ci.org/aspnet/KestrelHttpServer/jobs/392932264#L896-L915)?

```
 [xUnit.net 00:00:09.6604270]     TlsHandshakeRejectsTlsLessThan12 [FAIL]
  Failed   TlsHandshakeRejectsTlsLessThan12
RUNDOTNET : error Message:  [/home/travis/.dotnet/buildtools/korebuild/2.2.0-preview1-17087/KoreBuild.proj]
   System.IO.IOException : The reader completed without returning a frame.
  Stack Trace:
     at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.Http2.TlsTests.ReceiveFrameAsync(PipeReader reader) in /_/test/Kestrel.FunctionalTests/Http2/TlsTests.cs:line 103
     at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.Http2.TlsTests.WaitForConnectionErrorAsync(PipeReader reader, Boolean ignoreNonGoAwayFrames, Int32 expectedLastStreamId, Http2ErrorCode expectedErrorCode) in /_/test/Kestrel.FunctionalTests/Http2/TlsTests.cs:line 72
     at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.Http2.TlsTests.TlsHandshakeRejectsTlsLessThan12() in /_/test/Kestrel.FunctionalTests/Http2/TlsTests.cs:line 67
  --- End of stack trace from previous location where exception was thrown ---
  Standard Output Messages:
   | [0.001s] TestLifetime Information: Starting test TlsHandshakeRejectsTlsLessThan12 at 2018-06-15T23:03:39
   | [0.009s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting starting
   | [0.016s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting started
   | [0.016s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Loaded hosting startup assembly Sockets.FunctionalTests, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60
   | [0.047s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLEJ2GTA2RIJ" started.
RUNDOTNET : | [3.966s] Microsoft.AspNetCore.Server.Kestrel error : HTTP/2 over TLS was not negotiated on an HTTP/2-only endpoint. [/home/travis/.dotnet/buildtools/korebuild/2.2.0-preview1-17087/KoreBuild.proj]
   | [3.967s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLEJ2GTA2RIJ" sending FIN.
   | [4.184s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLEJ2GTA2RIJ" stopped.
   | [4.266s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting shutdown
   | [4.922s] TestLifetime Information: Finished test TlsHandshakeRejectsTlsLessThan12 in 4.9208921s
```